aten/src/ATen/templates/BackendSelectRegister.cpp

```diff
@@ -0,0 +1,22 @@
+// This is used for factory methods dispatching. We register ops with a high priority dispatch key,
+// which makes the dispatcher to dispatch here. We compute the dispatch key 'manually' and navigate to the
+// correct backend.
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <c10/core/TensorOptions.h>
+
+namespace at {
+namespace native {
+
+namespace {
```

Comment:
did you respond to why this whole file can't be in an anonymous namespace?  (i.e. why do you need the at::native:: namespace).

Reply:
I guess that's fine, but it's just namespace lookup that you can solve with a couple of "using" in the template.